src/ui/states/state-needs-payment-info.svelte

```diff
@@ -89,257 +122,170 @@
     }
   }
 
-  onMount(async () => {
-    const stripePk = paymentInfoCollectionMetadata.data.publishable_api_key;
-    const stripeAcctId = paymentInfoCollectionMetadata.data.stripe_account_id;
+  onMount(() => {
+    updateStripeVariables();
 
-    if (!stripePk || !stripeAcctId) {
-      throw new Error("Stripe publishable key or account ID not found");
-    }
+    window.addEventListener("resize", onResize);
 
-    stripe = await loadStripe(stripePk, {
-      stripeAccount: stripeAcctId,
-    });
+    return () => {
+      window.removeEventListener("resize", onResize);
+    };
+  });
+
+  onMount(() => {
+    let isMounted = true;
+
+    (async () => {
+      try {
+        const { stripe: stripeInstance, elements: elementsInstance } =
+          await StripeService.initializeStripe(
+            paymentInfoCollectionMetadata,
+            brandingInfo,
+            localeToUse,
+            stripeVariables,
+            viewport,
+          );
+
+        if (!isMounted) return;
+
+        stripe = stripeInstance;
+        elements = elementsInstance;
+
+        paymentElement = StripeService.createPaymentElement(
+          elements,
+          brandingInfo?.app_name,
+        );
+
+        paymentElement.mount("#payment-element");
+        paymentElement.on(
+          "change",
+          (event: StripePaymentElementChangeEvent) => {
+            isPaymentInfoComplete = event.complete;
+          },
+        );
+      } catch (error) {
+        console.error("Failed to initialize Stripe:", error);
+        modalErrorMessage = "Failed to initialize payment system";
```

Comment:
we might not want to surface this error to the client, and in fact we should redirect to the error view or something

Reply:
Fixed. A non-recoverable generic error will be surfaced.